tests/common.bash

```diff
@@ -1,15 +1,62 @@
 #!/bin/bash
 
-function clean_up_environment {
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found pods --all
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found -n kubewarden clusteradmissionpolicies --all
-	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found -n kubewarden admissionpolicies --all
-	kubectl --context $CLUSTER_CONTEXT wait --for=condition=Ready -n kubewarden pod --all
+bats_require_minimum_version 1.5.0
+
+function kubectl() {
+	command kubectl --context $CLUSTER_CONTEXT "$@"
+}
+
+function helm() {
+	command helm --kube-context $CLUSTER_CONTEXT "$@"
+}
+
+function retry() {
+    local cmd=$1
+    local tries=${2:-10}
+    local delay=${3:-30}
+    local i
+    for ((i=1; i<=tries; i++)); do
+        timeout 25 bash -c "$cmd" && break || echo "RETRY #$i: $cmd"
+        [ $i -ne $tries ] && sleep $delay || { echo "Godot: $cmd"; false; }
+    done
+}
+
+# Safe version of vaiting for pods
+# Handles kube-api disconnects during upgrade
+function wait_all_pods() {
+    local i output
+    for i in {1..90}; do
+        output=$(kubectl get pods --no-headers -n kube-system -o wide | grep -vw Completed || echo 'Fail')
+        grep -vE '([0-9]+)/\1 +Running' <<< $output || break
+        [ $i -ne 6 ] && sleep 30 || { echo "Godot: pods not running"; false; }
+    done
+}
+
+# Safe version of vaiting for nodes
+# Handles kube-api disconnects during upgrade
+function wait_all_nodes() {
+    local i output
+    for i in {1..30}; do
+        output=$(kubectl get nodes --no-headers || echo 'Fail')
+        grep -vE '\bReady\b' <<< $output || break
+        [ $i -ne 6 ] && sleep 30 || { echo "Godot: nodes not running"; false; }
+    done
+}
+
+function wait_pods   () { [ $# -gt 0 ] && kubectl wait pods --for=condition=ready --timeout=5m "$@" || wait_all_pods; }
+function wait_nodes  () { [ $# -gt 0 ] && kubectl wait nodes --for=condition=ready --timeout=5m "$@" || wait_all_nodes; }
+function wait_deploy () { kubectl wait deployment --for=condition=available --timeout=5m "$@"; }
+function wait_rollout() { kubectl rollout status  --timeout=5m "$@"; }
+
+# Wait for cluster to come up after reboot
+function wait_cluster() {
+    retry "kubectl cluster-info" 20 30
+    wait_nodes
+    wait_pods
 }
 
 function kubectl_apply_should_fail {
-	run kubectl --context $CLUSTER_CONTEXT apply --wait --timeout $TIMEOUT  -f $1
-	[ "$status" -ne 0 ]
+	run ! kubectl apply --wait --timeout $TIMEOUT  -f $1
```

Comment:
Took me a while to find the line in [the bats docs](https://bats-core.readthedocs.io/en/stable/writing-tests.html#run-test-other-commands) that says:
`! expect nonzero exit status (1-255), fail if command succeeds`

I would prefer to keep bats-related things to a minimum, and keep checking the status manually (it also helps when debugging, or copy-pasting bash code from the tests when performing manual tests).

Reply:
makes sense, I can revert this, I sometimes like minimal code too much :)